test/forge/TestCaseParser.sol

```diff
@@ -0,0 +1,341 @@
+        }
+        return strategies;
+    }
+
+    /**
+     * @dev helper function to parse test data json object to TradeAction[] struct
+     */
+    function parseTradeActions(
+        string memory json,
+        string memory templateName
+    ) private returns (TradeAction[] memory tradeActions) {
+        string memory initialParseString = string.concat("$.", templateName);
+        initialParseString = string.concat(initialParseString, ".tradeActions");
+
+        // read the trade actions length
+        string[] memory tradeActionsString = vm.parseJsonStringArray(json, initialParseString);
+        uint256 tradeActionsLength = tradeActionsString.length;
+
+        initialParseString = string.concat(initialParseString, "[");
+
+        // initialize trade actions array
+        tradeActions = new TradeAction[](tradeActionsLength);
+
+        for (uint i = 0; i < tradeActionsLength; ++i) {
+            // get the correct trade action index to parse
+            string memory parseString = string.concat(initialParseString, uintToString(i));
+
+            // Parse the trade actions field into bytes format
+            bytes memory tradeActionBytes = json.parseRaw(string.concat(parseString, "]"));
+            // Decode the bytes into an TradeActionIntermediate struct
+            TradeActionIntermediate memory tradeActionIntermediate = abi.decode(
+                tradeActionBytes,
+                (TradeActionIntermediate)
+            );
+            // Format the TradeAction struct properly
+            tradeActions[i] = TradeAction({
+                strategyId: bytesToUint(tradeActionIntermediate.strategyId),
+                amount: uint128(stringToUint((tradeActionIntermediate.amount)))
+            });
+        }
+        return tradeActions;
+    }
+
+    /// @dev convert an order struct to uint
+    function convertOrderStructToUint(OrderString memory orderString) private pure returns (Order memory order) {
+        return
+            Order({
+                y: uint128(stringToUint(orderString.y)),
+                z: uint128(stringToUint(orderString.z)),
+                A: uint64(stringToUint(orderString.A)),
+                B: uint64(stringToUint(orderString.B))
+            });
+    }
+
+    /// @dev helper function to convert a string to uint256
+    function stringToUint(string memory s) private pure returns (uint256 result) {
+        bytes memory b = bytes(s);
+        result = 0;
+        for (uint256 i = 0; i < b.length; i++) {
+            uint256 c = uint256(uint8(b[i]));
+            if (c >= 48 && c <= 57) {
+                result = result * 10 + (c - 48);
+            }
+        }
+        return result;
+    }
+
+    /// @dev helper function to convert an uint256 to string
+    function uintToString(uint256 v) private pure returns (string memory) {
+        if (v == 0) return "0";
+
+        uint256 maxlength = 100;
+        bytes memory reversed = new bytes(maxlength);
```

Comment:
```
bytes[100] memory reversed;
```

Reply:
Have to allocate the array dynamically in order to be able to populate it like so:
`reversed[i++] = bytes1(uint8(48 + remainder));`

Reply:
The bytes1 casting doesn't work using the static allocation. Anyway, I implemented the approach you suggested below.